cpp/src/arrow/table_test.cc

```diff
@@ -429,6 +429,32 @@ TEST_F(TestTable, CombineChunks) {
   }
 }
 
+TEST_F(TestTable, LARGE_MEMORY_TEST(CombineChunksStringColumn)) {
+  schema_ = schema({field("str", utf8())});
+  arrays_ = {nullptr};
+
+  constexpr int64_t kBinaryMemoryLimit = std::numeric_limits<int32_t>::max() - 1;
```

Comment:
This is declared in `array/builder_binary.h`. Either we should use that or move the declaration to somewhere like `arrow/type_fwd.h`